modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -4616,16 +4623,10 @@ public CacheGroupDescriptor cacheGroupDescriptor(int grpId) {
      * @return Cache descriptor.
      */
     public @Nullable DynamicCacheDescriptor cacheDescriptor(int cacheId) {
-        for (DynamicCacheDescriptor cacheDesc : cacheDescriptors().values()) {
-            CacheConfiguration ccfg = cacheDesc.cacheConfiguration();
-
-            assert ccfg != null : cacheDesc;
+        Map<Integer, DynamicCacheDescriptor> cacheIdToNameMapping = cacheIdToNameMapping();
```

Comment:
It seems this code can be a "one-liner": `return cacheIdToNameMapping().get(cacheId);`